sky/provision/runpod/utils.py

```diff
@@ -77,7 +77,11 @@ def list_instances() -> Dict[str, Dict[str, Any]]:
         info['name'] = instance['name']
         info['port2endpoint'] = {}
 
-        if instance['desiredStatus'] == 'RUNNING' and instance.get('runtime'):
+        # Sometimes when the cluster is in the process of being created,
+        # the `port` field in the runtime is None and we need to check for it.
+        if (instance['desiredStatus'] == 'RUNNING' and
+                instance.get('runtime') and
+                instance.get('runtime').get('ports')):
```

Comment:
In this case, do we need to retry until the port is available, instead of just skipping it? Otherwise, our provision may stuck?

Reply:
We actually already have such logic in the provision lib:

https://github.com/skypilot-org/skypilot/blob/c25d6eaf484ccc828fe944909a004450c175e859/sky/provision/runpod/instance.py#L100-L110